### DISCUSSION TITLE
Is it possible to somehow get the path to the installed package (for example httpd) using tea-cli?

### DISCUSSION THREAD

**Discussion by mfts** [pkgxdev/pkgx#564] [Help]
_asked via Discord by @ArionThinker_ 

Now I have found only such a solution  
```sh
$(tea --prefix)/apache.org/httpd/v*/
```

**Comment by davdroman** [comment]
I've wanted this as well on a couple occasions. Perhaps a `tea --which` option would be good?

**Reply by jhheider** [reply]
It seems like a good idea for disambiguation. Since `tea` tries to work within your shell environment, `tea +deno.land which deno` or `tea +deno.land command -v deno` will work, but what that _doesn't_ do is tell you what `tea`'s command-not-found magic will give you for a command if you don't know the package name (although `tea +tea.xyz/brewkit pkg search deno` will tell you that). So, the information is locatable today, but not as simple as it could be.

**Comment by mxcl** [comment]
I think some usage examples are required here. Are we trying to find out where a binary is, eg `httpd` or get the prefix for a specific package?

I think `tea --prefix httpd` makes sense.

While I think `tea --prefix gnu.org/wget` makes less sense since it's not much different from `$(tea --prefix)/gnu.org/wget`

But well if we do the former we'll do the later anyway. I guess I'm asking because I want to understand what specifically was desired to make sure we deliver it.

**Reply by mxcl** [reply]
FYI I'm suggesting `tea --prefix foo` as we already have `tea --prefix` and then this also mirrors how `brew --prefix` works.

**Comment by and-ri** [comment]
> I think some usage examples are required here. Are we trying to find out where a binary is, eg httpd or get the prefix for a specific package?
> 
> I think tea --prefix httpd makes sense.
> 
> While I think tea --prefix gnu.org/wget makes less sense since it's not much different from $(tea --prefix)/gnu.org/wget
> 
> But well if we do the former we'll do the later anyway. I guess I'm asking because I want to understand what specifically was desired to make sure we deliver it.

it would be mega cool if ‘tea --prefix httpd’ would return the path to the httpd installed via tea. If you're talking about it. Similar to {{deps.apache.org/httpd.prefix}} in package.yml

**Reply by jhheider** [reply]
The only wrinkle you might need or want to consider is that the prefix in a build script is fully resolved to a specific version, but there may be an infinite number present on the system. The logical choice is returning the same thing you'd get from a `tea httpd` magic invocation (the highest installed version number), but it makes me wonder if it needs (like `which`) an `--all` option.

This would be of particular interest for the same binary name provided by two different packages. We don't have that yet (I think), but you could imagine it for `telnet` or `cc`. The ability to find all of them instead whichever is "first" seems valuable for some integrators.

**Reply by mxcl** [reply]
I think it works, `tea --prefix httpd` returns latest unless we're in a dev-env. you can do `tea --prefix httpd^3` as well

**Reply by mxcl** [reply]
```sh
$ tea -n httpd
/Users/mxcl/.tea/apache.org/httpd/v2.4.57/bin/httpd
```

of course `--dry-run` already outputs this. It *might* output other stuff though, this isn't guaranteed to be parsable.

However `tea --prefix httpd` should only output `~/.tea/apache.org/httpd/v2.4.57` imo because that is the prefix. Hmm.

**Reply by jhheider** [reply]
Which argues for a --which or similar.

**Reply by mxcl** [reply]
We have `--provides` already we could output the binpath with that. This is an API change for anyone using it already, they'd want to do `--silent --provides` but that seems ok.
